app/controllers/dashboard_controller.rb

```diff
@@ -14,6 +14,16 @@ def index
     end
   end
 
+  def show
+    if @widget.present? && @widget.user == User.current
+      render(:partial => @widget.template, :locals => @widget.data)
```

Comment:
also, not sure if its related, but i got the following when trying to render the discovery widgets:
```
W] Failed to load widget
10:30:50 rails.1   |  | ActionView::MissingTemplate: Missing partial dashboard/_discovery_widget, application/_discovery_widget with {:locale=>[:en], :formats=>[:html, :html, :text, :js, :css, :ics, :csv, :vcf, :png, :jpeg, :gif, :bmp, :tiff, :mpeg, :xml, :rss, :atom, :yaml, :multipart_form, :url_encoded_form, :json, :pdf, :zip], :variants=>[], :handlers=>[:erb, :builder, :raw, :ruby, :rabl]}. Searched in:
10:30:50 rails.1   |  |   * "/home/ohad/git/foreman/app/views"
10:30:50 rails.1   |  |   * "/home/ohad/.rvm/gems/ruby-2.3.1@foreman/gems/apipie-rails-0.3.6/app/views"
10:30:50 rails.1   |  | 
```

note that the plugin path was not included?